packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/exception/CodedException.kt

```diff
@@ -247,6 +247,13 @@ internal class CollectionElementCastException private constructor(
   ) : this(collectionType, elementType, providedType.toString(), cause)
 }
 
+@DoNotStrip
+class DynamicCastException(
+  typeName: String
```

Comment:
maybe we can pass class type and get its string name rather than having the handy `map`, `string`, `array` as inputs

Reply:
Agree. Have changed this now!